### ISSUE TITLE
fail to compile on AlpineLinux

### ISSUE THREAD

**Issue by ogoudron** [jwbensley/Etherate#9]
Hello,
Etherate failed to compile on Alpine Linux (musl libc based distro) with this error mesage.
In file included from etherate.cpp:77:0:
etherate.h:164:21: error: field 'TV_SELECT_DELAY' has incomplete type 'timeval'
     struct  timeval TV_SELECT_DELAY;    // Elapsed time struct for polling the socket FD
                     ^~~~~~~~~~~~~~~
etherate.h:164:13: note: forward declaration of 'struct timeval'
     struct  timeval TV_SELECT_DELAY;    // Elapsed time struct for polling the socket FD
             ^~~~~~~
make: *** [Makefile:444: etherate.o] Error 1

**Comment by jwbensley** [comment]
Thanks for reporting the issue. I'll check it out when I get some time. This might not be fixable, due to the vast number of Linux distro's fixing this for Alpine might break it for Ubuntu for example. I personally am only interested in running Etherate on Ubuntu and CentOS (happy to make a new branch and someone can  submit a pull request though!). I'll see if there is a fix that doesn't hinder Ubunt and CentOS compiling and update when I know more.

**Comment by ogoudron** [comment]
Thanks for your reply. I think Etherate would be a nice tool to have for Alpine Linux.

**Comment by jwbensley** [comment]
I had every intention of fixing this issue :)

I will update the INSTALL documentation, I have it compiling now by using the following steps:

`apk add gcc automake autoconf libtool m4 alpine-sdk linux-headers`

Then either:
```
./configure.sh
make
```
Or:
```
gcc -o etherate etherate.c -Wall -Werror -ftrapv -pedantic -lrt -lm -std=gnu11 -O3
```

**Comment by ogoudron** [comment]
Hello, thanks for your answer.
FYI, i closed this ticket because i switched to VoidLinux. Compilation worked on VoidLinux but i have not done yet all tests.

**Comment by jwbensley** [comment]
This is fixed version 1.17 (2018-04) - I have updated the INSTALL file for any future users of Alpine with the install requirements.
